src/openloco/windows/constructionwnd.cpp

```diff
@@ -1,75 +1,4836 @@
+            if (currentYear < bridgeObj->designed_year)
+                continue;
+            _bridgeList[bridgeCount] = bridge;
+            bridgeCount++;
+        }
+
+        for (uint8_t i = 0; i < objectmgr::get_max_objects(object_type::bridge); i++)
+        {
+            auto bridgeObj = objectmgr::get<bridge_object>(i);
+            if (bridgeObj == nullptr)
+                continue;
+            for (auto j = 0; j < bridgeObj->var_22; j++)
+            {
+                if (trackType != bridgeObj->var_23)
+                    continue;
+                if (currentYear < bridgeObj->designed_year)
+                    continue;
+                for (size_t k = 0; k < std::size(_bridgeList); k++)
+                {
+                    auto bridge = bridgeListCopy[k];
+                    if (bridgeListCopy == _bridgeList)
+                    {
+                        _bridgeList[bridgeCount] = i;
+                        bridgeCount++;
+                        break;
+                    }
+                    if (i == bridge)
+                        break;
+                }
+            }
+        }
+
+        _bridgeList[bridgeCount] = 0xFF;
+
+        sortList(_bridgeList, std::size(_bridgeList));
+    }
+
+    // 0x004781C5
+    static void refreshRoadModList()
+    {
+        auto trackType = _trackType & ~(1 << 7);
+        auto companyId = _updatingCompanyId;
+        _modList[0] = 0xFF;
+        _modList[1] = 0xFF;
+        _modList[2] = 0xFF;
+        _modList[3] = 0xFF;
+        auto flags = 0;
+
+        for (uint8_t vehicle = 0; vehicle < objectmgr::get_max_objects(object_type::vehicle); vehicle++)
+        {
+            auto vehicleObj = objectmgr::get<vehicle_object>(vehicle);
+
+            if (vehicleObj == nullptr)
+                continue;
+
+            if (vehicleObj->mode != TransportMode::road)
+                continue;
+
+            if (trackType != vehicleObj->track_type)
+                continue;
+
+            auto edi = vehicle;
+            auto edx = vehicle & 0x1F;
+            edi = edi >> 5;
+
+            auto company = companymgr::get(companyId);
+
+            if (!(company->unlocked_vehicles[edi] & 1 << edx))
+                continue;
+
+            for (auto i = 0; i < vehicleObj->num_mods; i++)
+            {
+                flags |= 1ULL << vehicleObj->required_track_extras[i];
+            }
+
+            if (!(vehicleObj->flags & flags_E0::rack_rail))
+                continue;
+
+            flags |= 1ULL << vehicleObj->rack_rail_type;
+        }
+
+        auto roadObj = objectmgr::get<road_object>(trackType);
+
+        for (auto i = 0; i < roadObj->num_mods; i++)
+        {
+            if (flags & (1 << roadObj->mods[i]))
+                _modList[i] = roadObj->mods[i];
+        }
+    }
+
+    // 0x004A3A50
+    static void sub_4A3A50()
+    {
+        common::sub_49FEC7();
+        setTrackOptions(_trackType);
+        refreshRoadStationList();
```

Comment:
I would prefer if this function still took its two variables (the list and the type)